frame/authority-discovery/src/lib.rs

```diff
@@ -124,10 +141,42 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 	{
 		// Remember who the authorities are for the new and next session.
 		if changed {
-			let keys = validators.map(|x| x.1);
-			Keys::<T>::put(keys.collect::<Vec<_>>());
-			let next_keys = queued_validators.map(|x| x.1);
-			NextKeys::<T>::put(next_keys.collect::<Vec<_>>());
+			let mut keys = validators.map(|x| x.1).collect::<Vec<_>>();
+
+			if keys.len() >= T::MaxAuthorities::get() as usize {
+				// Truncate to MaxAuthorities, to not fail the conversion
+				keys.truncate(T::MaxAuthorities::get() as usize);
```

Comment:
similarly to https://github.com/paritytech/substrate/pull/9371 PR I would prefer to use WeakBoundedVec here. So that instead of truncating some authorities, resulting in some probably quite annoying result, we can only have a slight underestimate of PoV.
The WeakBoundedVec will log a warning. and any correctly configured chain should never overflow this weak bound.